chalicelib_fourfront/checks/wfr_checks.py

```diff
@@ -147,10 +149,26 @@ def md5run_status(connection, **kwargs):
     not_switched_status_to_wait = []
     # multiple failed runs
     problems = []
-    my_s3_util = s3Utils(env=connection.ff_env)
-    raw_bucket = my_s3_util.raw_file_bucket
-    out_bucket = my_s3_util.outfile_bucket
-    for a_file in res:
+    # 2024-02-08: Per above comment WRT any() on a generator, only do
+    # this S3 access only if we have at least one result in the loop below.
+    # my_s3_util = s3Utils(env=connection.ff_env)
+    # raw_bucket = my_s3_util.raw_file_bucket
+    # out_bucket = my_s3_util.outfile_bucket
+    my_s3_util = None
+    while True:
+        try:
+            # TODO: Be nicer to have a generator wrapper to allow easily
+            # checking for any results without skipping past the first item.
+            a_file = next(res)
```

Comment:
I don’t really like the way I did this, but best i could do without being more disruptive of the code for short term - though let me know what you think, if you have a better idea.

Reply:
You can do this by making a second request with limit=0 and checking the "total" field in the (non-generator) response, and if there are results, then request the generator, otherwise exit

Reply:
I had initially thought to just get rid of the generator all together but we have a submitter just about to upload more than 1000 files that will need runs so best to either keep it the way you have it or use Will's approach.